### ISSUE TITLE
Port to C#

### ISSUE THREAD

**Issue by dajuric** [FSCL/FSCL.Compiler#13]
Hi,

  first I wanted to tank you for the awesome library!
I have a bare-bones image platform written in C# [DotImaging](https://github.com/dajuric/dot-imaging) that could benefit from OpenCL (its primitive is .NET array).

I was wondering how much effort is needed to convert/enable this library for C# ?

Thanks!

**Comment by Kjue** [comment]
Having taken a look into the library a bit and knowing that it behaves nicely with CLR of any kind, you could conceivably use it already with C#, and you can. Thing is, you'll need to write your code that you want to run in OpenCL in F#, because the translator at the moment does not handle other languages. It should also be possible to write up the conversions for C# as well, but I have no idea about the amount of work it would detail. 

I was checking out your library at some point to use with FSCL as well, and I was running through some prototype code for it, and I believe it should be doable to hook up an imaging library in such manner, that it would be easy to handle in CL-code too. Currently no such functionality exists in this library, and I think that is among the future plans for the library.

**Comment by dajuric** [comment]
Many thanks :)
I really hope you will maintain and extend the library. In fact I think you
should 'advertise' it more (e.g. by writing an article on CodeProject),
because there is a huge potential.

Best,
Darko

2016-07-07 8:09 GMT+02:00 Kjue notifications@github.com:

**Comment by smoothdeveloper** [comment]
@dajuric, I don't think this is feasible, this project relies on F# code quotations (which is something similar to LINQ Expr in general .NET).

Options: 
* make a LINQ Expr to F# code quotation translation
* contribute to this project so components of it can be reused at the stage after processing the F# quotations to intermediary structure and implement another front end for LINQ Expr to that intermediary structure

I think we should close this issue.

**Comment by dajuric** [comment]
Thanks.

2017-01-26 17:11 GMT+01:00 Gauthier Segay <notifications@github.com>:
